### DISCUSSION TITLE
Ollama Web UI crashing when uploading files to RAG

### DISCUSSION THREAD

**Discussion by jannikstdl** [open-webui/open-webui#541] [General]
# Bug Report
Hi, i have set up the project with kubernetes on my cluster. I am not using helm charts. 
When uploading files to RAG the Pod crashes.
## Description
After the crash the Pod restarts as usual, but all data including the registred users are lost.

**Bug Summary:**
Ollama Web UI crashing when uploading files to RAG

**Steps to Reproduce:**
- Kubernetes Deployment of the Project
- Tested RAG with PDF

**Expected Behavior:**
Document is loading as usual, like on my local machine.

**Actual Behavior:**
After adding the file (using the method in the chat input and over the sidebar under "documents") The File upload keeps loading and after a few seconds the pod crashes.
## Environment

- **Operating System:** Linux (Kubernetes Cluster)
- **Browser (if applicable):** [Edge latest]

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I have reviewed the troubleshooting.md document.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
![Screenshot 2024-01-20 203832](https://github.com/ollama-webui/ollama-webui/assets/69747628/3c732562-71df-478e-85b2-2070a21fc2cc)

**Docker Container Logs:**
Ollama Web UI Pod logs:
INFO:     100.127.42.247:0 - "GET /themes/rosepine.css HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /api/v1/ HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /themes/rosepine-dawn.css HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /api/v1/auths/ HTTP/1.1" 401 Unauthorized
INFO:     100.127.42.247:0 - "GET /_app/immutable/nodes/13.f487568d.js HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /_app/immutable/assets/13.e43bb62b.css HTTP/1.1" 200 OK
insert_new_auth
INFO:     100.127.42.247:0 - "POST /api/v1/auths/signup HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /_app/immutable/nodes/3.7011202d.js HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /_app/immutable/assets/index.e3b04202.css HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /_app/immutable/chunks/index.4da1d489.js HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /api/v1/modelfiles/ HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /api/v1/prompts/ HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /api/v1/documents/ HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /api/v1/chats/tags/all HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /ollama/api/version HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /ollama/api/tags HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /ollama/api/url HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
https://api.openai.com/v1/models
INFO:     100.127.42.247:0 - "GET /openai/api/models HTTP/1.1" 401 Unauthorized
INFO:     100.127.42.247:0 - "GET /openai/api/url HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /openai/api/key HTTP/1.1" 200 OK
INFO:     100.127.42.247:0 - "GET /ollama/api/version HTTP/1.1" 200 OK
application/pdf
Killed

## Installation Method

Kubernetes

**Comment by tjbck** [comment]
Hi, Thanks for creating this issue. I'll move this post to discussion as it does not pertain to any of the core functionalities of the webui. I personally have not tested the webui with k8s, maybe @dnviti could help you out with the troubleshooting.

**Comment by jannikstdl** [comment]
Hey @dnviti any thoughts on this?

**Reply by dnviti** [reply]
Hello! I'll check that as soon as possible!
I think that it happens because there is no storage persistance, which is demanded to the user and must be based on the usrs's k8s configuration (for example an NFS mount)

**Reply by jannikstdl** [reply]
Ok we managed to fix this. This was happening because the Ollama WebUI yaml (ollama-deployment.yaml) has resource limits of too small allocated memory. When the Files got vectorized the pod ran out of memory. Don't know exactly if the SyntaxError <html... Error was the cause or just a symptome of that, but increasing the limits or removing them completely helped. 

We have also added a PVC to the /backend/data path but I don't think that has something to to with that.

**Reply by dnviti** [reply]
Oh yes, current resource limits are only there set for the sake of the example template, both resource limits and storage capacity should be changed by the user accordingly 😁 .
The same applies to Helm's values.yaml.

**Comment by pkaramertzanis** [comment]
Where can we change the resources in the latest version of oepn-webui?
